.github/workflows/scripts/slack_gem_notifications/notifications_methods.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require 'time'
+require "httparty"
+
+def check_for_updates(watched_gems)
+  return if gem_list_empty?(watched_gems)
+  watched_gems.each do |gem_name|
+    return unless is_valid_gem?(gem_name)
+    versions = get_gem_info(gem_name)
+    send_bot(gem_name, versions) if gem_updated?(versions)
+  end
+end
+
+def gem_list_empty?(watched_gems)
+  if watched_gems.empty?
+    raise 'Nothing to see here! watched_gems cannot be empty'
+    exit
+  end
+
+  watched_gems.empty?
+end
+
+def is_valid_gem?(gem_name)
+  res = HTTParty.get("https://rubygems.org/api/v1/versions/#{gem_name}.json")
+
+  res.code == 200
+end
+
+# Gem entries are ordered by release date. The break limits versions to two versions: newest and previous.
+# TODO: If possible, specify how many gem objects we recieve from Ruby Gems API.
+def get_gem_info(gem_name)
+  gem_info = HTTParty.get("https://rubygems.org/api/v1/versions/#{gem_name}.json")
+  versions = []
+  gem_info.each do |gem|
+    versions << gem if gem['platform'] == 'ruby'
+    break if versions.size == 2
+  end
+
+  versions
+end
```

Comment:
What happened to our refactor of this method?

Reply:
I  couldn't get the refactor to work :( We added a slice but I realized that doesn't work because we first need to grab two items that have gem['platform'] == 'ruby'.

I'm interested in making this smoother but have run out of ideas.

Reply:
We want to grab the first 2 hashes in the `gem_info` array whose `'platform'` values are `'ruby'`.

If we were to use `#map` or `#select`, we run into inefficiencies because we have to check the `'platform'` value for every single hash and only afterwards cut down the list to `2`.

```ruby
# inefficient - checks on every entry in the array and then grabs 2 afterwards
gem_info.select { |gem| gem['platform'] == 'ruby' }.take(2)
```

Kayla's original comment pointed to not having to declare an empty array and then fill it as two separate steps. We can shrink things down with `#each_with_object` and still retain the efficiency that comes with an early break. It just looks a little different, because `break` in that context will need a return value:

```ruby
versions = gem_info.each_with_object([]) do |gem, arr|
  arr << gem if gem['platform'] == 'ruby'
  break arr if arr.size == 2
end
```